src/v2/Apps/Artists/Routes/ArtistsByLetter.tsx

```diff
@@ -0,0 +1,142 @@
+import { Box, Text, media, space } from "@artsy/palette"
+import React, { useState } from "react"
+import { graphql, createRefetchContainer, RelayRefetchProp } from "react-relay"
+import styled, { css } from "styled-components"
+import { RouterLink } from "v2/Artsy/Router/RouterLink"
+import { useRouter } from "v2/Artsy/Router/useRouter"
+import { PaginationFragmentContainer } from "v2/Components/Pagination"
+import { ArtistsTopNav } from "../Components/ArtistsTopNav"
+import { ArtistsByLetter_viewer } from "v2/__generated__/ArtistsByLetter_viewer.graphql"
+import { ArtistsByLetterMeta } from "../Components/ArtistsByLetterMeta"
+
+const Columns = styled(Box)<{ isLoading: boolean }>`
+  column-count: 4;
+  ${media.md`column-count: 3;`}
+  ${media.xs`column-count: 1;`}
+
+  ${({ isLoading }) =>
+    isLoading &&
+    css`
+      opacity: 0.25;
+      pointer-events: none;
+    `}
+`
+
+const Name = styled(RouterLink)`
+  display: block;
+  text-decoration: none;
+  padding: ${space(0.5)}px 0;
+`
+
+interface ArtistsByLetterProps {
+  viewer: ArtistsByLetter_viewer
+  relay: RelayRefetchProp
+}
+
+export const ArtistsByLetter: React.FC<ArtistsByLetterProps> = ({
+  relay,
+  viewer: {
+    artistsConnection: {
+      artists,
+      pageCursors,
+      pageInfo: { endCursor, hasNextPage },
+    },
+  },
+}) => {
+  const { match } = useRouter()
+  const [isLoading, setLoading] = useState(false)
+
+  const handleNext = () => {
+    handleClick(endCursor)
+  }
+
+  const handleClick = (cursor: string) => {
+    setLoading(true)
+
+    relay.refetch({ after: cursor }, null, error => {
+      if (error) {
+        console.error(error)
+      }
+
+      setLoading(false)
+    })
+  }
+
+  return (
+    <>
+      <ArtistsByLetterMeta />
+
+      <Box>
+        <Text my={3}>
+          <RouterLink to="/artists2" noUnderline>
+            Artists
+          </RouterLink>{" "}
+          / Browse all 50,000 artists
+        </Text>
+
+        <ArtistsTopNav my={3}>
+          <Text as="h1" variant="largeTitle">
+            Artists
+            {match.params.letter && <> – {match.params.letter.toUpperCase()}</>}
+          </Text>
+        </ArtistsTopNav>
+
+        <Columns my={3} isLoading={isLoading}>
+          {artists.map(({ artist }) => {
+            return (
+              <Text key={artist.internalID}>
+                <Name to={artist.href}>{artist.name}</Name>
+              </Text>
+            )
+          })}
+        </Columns>
+
+        <PaginationFragmentContainer
+          hasNextPage={hasNextPage}
+          pageCursors={pageCursors}
+          onNext={handleNext}
+          onClick={handleClick}
```

Comment:
API wise these do the same thing -- any reason we can't consolidate?

Reply:
They don't really do the same thing, onNext pages a single time, onClick accepts a cursor and pages.